src/test/java/com/github/webicitybrowser/spec/fetch/FetchEngineTest.java

```diff
@@ -1,39 +1,71 @@
 package com.github.webicitybrowser.spec.fetch;
 
+import com.github.webicitybrowser.spec.fetch.TaskDestination.ParallelQueue;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
 import com.github.webicitybrowser.spec.fetch.builder.FetchParametersBuilder;
-import com.github.webicitybrowser.spec.fetch.builder.FetchRequestBuilder;
 import com.github.webicitybrowser.spec.fetch.builder.FetchResponseBuilder;
 import com.github.webicitybrowser.spec.fetch.connection.FetchConnection;
 import com.github.webicitybrowser.spec.fetch.connection.FetchConnectionInfo;
 import com.github.webicitybrowser.spec.fetch.connection.FetchConnectionPool;
 import com.github.webicitybrowser.spec.fetch.imp.FetchEngineImp;
 import com.github.webicitybrowser.spec.url.URL;
 
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.stream.IntStream;
+
+import static org.mockito.ArgumentMatchers.eq;
```

Comment:
We are not using static imports
Also, use Mockito.eq instead of ArgumentMatchers.eq